master/internal/api_config_policies.go

```diff
@@ -74,21 +82,70 @@ func (*apiServer) GetGlobalConfigPolicies(
 }
 
 // Delete workspace task config policies.
-func (*apiServer) DeleteWorkspaceConfigPolicies(
+func (a *apiServer) DeleteWorkspaceConfigPolicies(
 	ctx context.Context, req *apiv1.DeleteWorkspaceConfigPoliciesRequest,
 ) (*apiv1.DeleteWorkspaceConfigPoliciesResponse, error) {
+	curUser, _, err := grpcutil.GetUser(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	err = workspace.AuthZProvider.Get().CanGetWorkspaceID(
```

Comment:
If a user doesn't have permissions to access a workspace, I don't think they should have the permissions to delete its config policies. 
I suppose though that the `CanModifyConfigPolicies` authz check is sufficient though, since I'm assuming any permission that grants a user access to a workspace's config policies is part of the same role that grants permissions to access the workspace.
I'll double check about this though!

Reply:
Removed the `CanGetWorkspaceID`check since every role that gives user the perms to update the config policies for a given scope also has the perms to modify the workspace in that scope, so this check is redundant. Good catch!